src/main/java/seedu/address/model/person/ReminderManager.java

```diff
@@ -3,92 +3,175 @@
      * @param persons ObservableList of {@code Person} from the address book.
      */
     public ReminderManager(ObservableList<Person> persons) {
         this.persons = persons;
+        setupChangeListener();
+        updateReminder(); // Initial update
     }
 
-    /**
-     * Fetches the latest reminders based on the nearest deadlines of persons in the list.
-     * This method will gather all persons who share the nearest upcoming deadline.
-     * @return A list of reminder messages, or an empty list if no reminders are due.
-     */
-    public List<String> getLatestReminders() {
-        LocalDate now = LocalDate.now();
-        List<String> reminders = new ArrayList<>();
-        LocalDate latestDeadline = null;
+    private void setupChangeListener() {
+        persons.addListener((ListChangeListener<Person>) change -> {
+            while (change.next()) {
+                if (change.wasAdded() || change.wasRemoved() || change.wasUpdated()) {
+                    updateReminder();
+                    break;
+                }
+            }
+        });
+    }
 
-        for (Person person : persons) {
-            Deadline deadline = person.getDeadline();
-            if (!deadline.isOverdue()) {
-                LocalDate deadlineDate = deadline.value;
+    private void updateReminder() {
+        List<String> reminders = getLatestReminders();
+        currentReminder.set(reminders.isEmpty() ? MESSAGE_NO_REMINDERS : reminders.get(0));
+    }
 
-                if (latestDeadline == null || deadlineDate.isBefore(latestDeadline)) {
-                    latestDeadline = deadlineDate;
-                    reminders.clear();
-                }
+    public StringProperty currentReminderProperty() {
+        return currentReminder;
+    }
 
-                if (deadlineDate.equals(latestDeadline)) {
-                    if (deadlineDate.isEqual(now)) {
-                        reminders.add(person.getName() + DUE_TODAY);
-                    } else {
-                        long daysUntilDeadline = ChronoUnit.DAYS.between(now, deadlineDate);
-                        String dayMessage = daysUntilDeadline == 1 ? DAY_SINGULAR : daysUntilDeadline + DAY_PLURAL;
-                        reminders.add(person.getName() + MESSAGE_DEADLINE_IN + dayMessage + PERIOD);
-                    }
+    private String formatNames(List<Person> persons) {
+        // Filter out completed projects
+        List<Person> incompletePeople = persons.stream()
+                .filter(person -> !person.getProjectStatus().isComplete)
+                .collect(Collectors.toList());
+
+        if (incompletePeople.isEmpty()) {
+            return "";
+        }
+
+        if (incompletePeople.size() == 1) {
+            return incompletePeople.get(0).getName().toString();
+        }
+
+        StringBuilder names = new StringBuilder();
+        int namesToShow = Math.min(incompletePeople.size(), MAX_NAMES_TO_SHOW);
+
+        for (int i = 0; i < namesToShow; i++) {
```

Comment:
Logic for this for loop in the case of the first if condition doesn't seem to match. If `namestoshow = 3`, and eg. `incompletepeople.size() = 5` i think string builder only has `2 names and 3 more` but I think it should be `3 names and 2 more`.

This is what i infer about max_names_to_show so correct me if i am wrong

Reply:
The current logic does result in showing only 2 names followed by "and 3 more" when MAX_NAMES_TO_SHOW is 3 and the list size is 5. I'll correct the logic to subtract the number of names shown from the total list size. Thanks for pointing that out!